lib/purchases_flutter.dart

```diff
@@ -581,20 +600,19 @@ class Purchases {
   ///
   /// [discount] The `Discount` to apply to the product.
   static Future<PaymentDiscount> getPaymentDiscount(
-      Product product, Discount discount) async {
-    Map<dynamic, dynamic> result =
-        await _channel.invokeMethod('getPaymentDiscount', {
+    Product product,
+    Discount discount,
+  ) async {
+    final result = await _channel.invokeMethod('getPaymentDiscount', {
       'productIdentifier': product.identifier,
       'discountIdentifier': discount.identifier
     });
-    return PaymentDiscount.fromJson(result);
+    return PaymentDiscount.fromJson(Map<String, dynamic>.from(result));
   }
 
   /// Android only. Call close when you are done with this instance of Purchases to disconnect
   /// from the billing services and clean up resources
-  static Future<void> close() async {
-    await _channel.invokeMethod('close');
-  }
+  static Future<void> close() => _channel.invokeMethod('close');
```

Comment:
do these not need `async` when using `=>`?

Reply:
not really, because `_channel.invokeMethod` returns `Future<T>` and we don't need `await` it. Only caller of method `close` can `await` it.